File: Depth-First Search/Sum Root to Leaf Numbers/solve.java
class Solution {
/*
 * Time Complexity: O(V) - DFS using recursion
 * Space Complexity: O(V) - create new String for every new node visited
 * Problem statement here: https://leetcode.com/problems/sum-root-to-leaf-numbers/
 * Do the basic DFS, keep a String at every node,
 * we add the value of that node to the String forming number.
 * When reached the leaf node, convert final number from String to Integer and add up to gobal variable.
 */
    private int currentSum;
    final private void dfsum(TreeNode ro, String currentStringNum){
        String newStringNum = currentStringNum.concat( String.valueOf(ro.val) );
        
        if( ro.left==null && ro.right==null ) {
            currentSum += Integer.parseInt(newStringNum);
            return;
        }
        
        if( ro.left != null )
            { dfsum(ro.left, newStringNum); }
        if( ro.right != null )
            { dfsum(ro.right, newStringNum); }
    }
    public int sumNumbers(TreeNode root) {
        this.currentSum=0;

        dfsum(root, "");
        return this.currentSum;
    }
}

File: Depth-First Search/Symmetric Tree/solve.cs
public class Solution {
    /*
    * Time Complexity: O(V+E)
    * Space Complexity: O(E) - due to stack
    */
    private bool IsSymmetricRecursive(TreeNode left, TreeNode right) {
        if(left is null && right is null) {
            return true;
        }

        if(left is null || right is null) {
            return false;
        }

        if(left.val != right.val) {
            return false;
        }

        return IsSymmetricRecursive(left.left, right.right)
                && IsSymmetricRecursive(left.right, right.left);
    }
    public bool IsSymmetric(TreeNode root) {
        TreeNode leftTraveller = root,
                    rightTraveller = root;

        return IsSymmetricRecursive(leftTraveller, rightTraveller);
    }
}